### PR TITLE
Added documentation examples for `consider-merging-isinstance`.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Part of https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

File: doc/data/messages/c/consider-merging-isinstance/bad.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class Magenta(Colour):
+    ...
+
+
+def is_primary_color(colour: Colour) -> bool:
+    return (
+        isinstance(colour, Red) or isinstance(colour, Blue) or isinstance(colour, Green)
+    )  # [consider-merging-isinstance]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/consider-merging-isinstance/bad.py:24]:
```suggestion
from typing import Any

def is_number(value: Any) -> bool:
    return isinstance(value, int) or isinstance(value, float)  # [consider-merging-isinstance]
```

File: doc/data/messages/c/consider-merging-isinstance/good.py [hunk 2]
```diff
@@ -1 +1,22 @@
...
+class Blue(Colour):
+    ...
+
+
+class Magenta(Colour):
+    ...
+
+
+def is_primary_color(colour: Colour) -> bool:
+    return isinstance(colour, (Red, Blue, Green))
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/consider-merging-isinstance/good.py:22]:
```suggestion
from typing import Any

def is_number(value: Any) -> bool:
    return isinstance(value, (int, float))
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think it's possible to simplify by using builtin type.

**Comment by mbyrnepr2** [comment]
> I think it's possible to simplify by using builtin type.
Much better 😄

**Comment by mbyrnepr2** [comment]
pre-commit.ci autofix

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**